torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -244,20 +244,12 @@ class ProcessGroupNCCL : public ProcessGroup {
     // synchronizing FutureNCCL's own cudaEvents with the stream that runs
     // this callback. This new FutureNCCL's cudaEvents will record the
     // callback's stream and will have the result value of the callback.
-    void addCallbackWithStream(
-        std::function<void(void)> callback,
-        const c10::cuda::CUDAStream& stream) {
-      (*cudaEvents_)[0].block(stream);
+    void addCallback(std::function<void(void)> callback) override {
+      (*cudaEvents_)[0].block(*futureNCCLCallbackStream_);
+      c10::OptionalStreamGuard streamGuard{c10::Stream(*futureNCCLCallbackStream_)};
       callback();
```

Comment:
Can we move these changes to https://github.com/pytorch/pytorch/pull/43447 instead of having it as part of this PR?

Reply:
Oh, probably pushed my local changes by mistake. Sorry, will fix that.